qutebrowser/utils/utils.py

```diff
@@ -864,6 +864,63 @@ def unused(_arg):
     pass
 
 
+def parse_numsettxt_into_list(txtset, nummax):
```

Comment:
You can probably save some time (and space!) by using [`itertools.chain`](https://docs.python.org/3.6/library/itertools.html#itertools.chain) (or `chain_from_iterable`) instead of eagerly creating a list with all numbers. `range` is already lazy on Python 3, so you just need to `chain` those ranges together to get a lazy result. You could probably also work with `yield from` instead of `chain`, whichever you prefer (`yield from range(int(start), int(end)+1)` should do the trick).

Also, I don't think it's necessary to squeeze out the *very best* asymptotic runtime. The numbers we're dealing with here are small (`O(number of downloads)` is probably less than 100 in most cases). In this case I think readable code is more important. And if we notice that this function really is a performance bottleneck (I doubt it), we can always profile, benchmark and optimize later.